From 8b8948c06de55455b63dc11f765d7ee39f30c4e0 Mon Sep 17 00:00:00 2001
From: Andreas Bombe <aeb@debian.org>
Date: Mon, 29 Dec 2014 18:24:54 +0100
Subject: [PATCH 03/14] mkfs.fat.c: Use unsigned char for binary data

Simple char technically works for the dummy_boot_jump variables, but
some compiler warning settings would give a warning over a signed char
overflowing with the values given as initializers.

Signed-off-by: Andreas Bombe <aeb@debian.org>
---
 src/mkfs.fat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mkfs.fat.c b/src/mkfs.fat.c
index 604b7d0..1f702ad 100644
--- a/src/mkfs.fat.c
+++ b/src/mkfs.fat.c
@@ -219,9 +219,9 @@ struct msdos_dir_entry {
 /* The "boot code" we put into the filesystem... it writes a message and
    tells the user to try again */
 
-char dummy_boot_jump[3] = { 0xeb, 0x3c, 0x90 };
+unsigned char dummy_boot_jump[3] = { 0xeb, 0x3c, 0x90 };
 
-char dummy_boot_jump_m68k[2] = { 0x60, 0x1c };
+unsigned char dummy_boot_jump_m68k[2] = { 0x60, 0x1c };
 
 #define MSG_OFFSET_OFFSET 3
 char dummy_boot_code[BOOTCODE_SIZE] = "\x0e"	/* push cs */
-- 
1.9.1

